### PR TITLE
add os x app categoryType option

### PR DESCRIPTION
**Pull request by plrthink**

### CODE REVIEW

**Comment by plrthink** [comment]
#200

**Comment by malept** [comment]
Thanks for the contribution. Could you add some docs to `usage.txt`, and add a test for this new option?

**Comment by plrthink** [comment]
@malept Sure, I would do that shortly.

**Comment by plrthink** [comment]
@malept added a test and docs to usage

**Comment by plrthink** [comment]
and upload this image to show it sets the application category type

screen shot 2015-11-21 at 13 08 33

File: usage.txt [hunk 1]
```diff
@@ -26,6 +26,9 @@ out                the dir to put the app into at the end. defaults to current w
 overwrite          if output directory for a platform already exists, replaces it rather than skipping it
 prune              runs `npm prune --production` on the app
 sign               should contain the identity to be used when running `codesign` (only for building for the darwin platform, on OS X)
+appCategoryType    set application category type.
```
Comment (malept) [inline usage.txt]: This is the only command line parameter that's camel cased. Is it possible for it to be hyphenated instead?

Comment (plrthink) [inline usage.txt]: Sure, sorry for breaking the code style... I would change that.

File: readme.md [hunk 2]
```diff
@@ -169,6 +169,13 @@ If the file extension is omitted, it is auto-completed to the correct extension
 
 `sign` - *String*
 
+`appCategoryType` - *String*
+
+  Application category type, showed by clicking through *Finder: View-->Arrange by Application Category* in *Application* directory (OS X only)
+
+  For example, `appCategoryType=public.app-category.developer-tools` will set app in *Developer Tools* category.
+  More allowed values see [Apple Docs](https://developer.apple.com/library/ios/documentation/General/Reference/InfoPlistKeyReference/Articles/LaunchServicesKeys.html#//apple_ref/doc/uid/TP40009250-SW8)
```
Comment (malept) [inline readme.md]: How's this?

> The application category type, as shown in the Finder via *View -> Arrange by Application Category* when viewing the `Applications` directory (OS X only).
>
>  For example, `appCategoryType=public.app-category.developer-tools` will set the application category to *Developer Tools*.
> Valid values are listed in [Apple's documentation](https://developer.apple.com/library/ios/documentation/General/Reference/InfoPlistKeyReference/Articles/LaunchServicesKeys.html#//apple_ref/doc/uid/TP40009250-SW8).

File: usage.txt [hunk 3]
```diff
@@ -26,6 +26,8 @@ out                the dir to put the app into at the end. defaults to current w
 overwrite          if output directory for a platform already exists, replaces it rather than skipping it
 prune              runs `npm prune --production` on the app
 sign               should contain the identity to be used when running `codesign` (only for building for the darwin platform, on OS X)
+app-category-type  the application category type (darwin platform only)
+                   For example, `appCategoryType=public.app-category.developer-tools` will set the application category to 'Developer Tools'.
```
Comment (malept) [inline usage.txt]: I think this line needs to be updated since you updated the CLI parameter name.

File: usage.txt [hunk 4]
```diff
@@ -26,6 +26,8 @@ out                the dir to put the app into at the end. defaults to current w
 overwrite          if output directory for a platform already exists, replaces it rather than skipping it
 prune              runs `npm prune --production` on the app
 sign               should contain the identity to be used when running `codesign` (only for building for the darwin platform, on OS X)
+app-category-type  the application category type (darwin platform only)
```
Comment (malept) [inline usage.txt]: Add a period at the end, for consistency.

File: readme.md [hunk 5]
```diff
@@ -169,6 +169,14 @@ If the file extension is omitted, it is auto-completed to the correct extension
 
 `sign` - *String*
 
+`app-category-type` - *String*
+
+  The application category type, as shown in the Finder via *View -> Arrange by Application Category* when viewing the Applications directory (OS X only).
+
+  For example, `appCategoryType=public.app-category.developer-tools` will set the application category to *Developer Tools*.
```
Comment (malept) [inline readme.md]: See: https://github.com/maxogden/electron-packager/pull/202/files#r46084226

**Comment by plrthink** [comment]
@malept check again~

**Comment by malept** [comment]
@plrthink :+1: looks good. Last thing is to rebase & squash the commits, and I'll merge.

**Comment by plrthink** [comment]
@malept Ok, I would do this ASAP.

**Comment by plrthink** [comment]
@malept Please check again~

**Comment by malept** [comment]
I just realized that the docs are in the wrong place (it's ordered by the parameter name, alphabetically), but I'll fix that post-merge, since I should have seen that earlier.

Thanks for your contribution!

**Comment by plrthink** [comment]
:+1:

**Pull request merged**